src/models/connectionStore.ts

```diff
@@ -202,9 +202,9 @@ export class ConnectionStore {
 	 * @param {forceWritePlaintextPassword} whether the plaintext password should be written to the settings file
 	 * @returns {Promise<IConnectionProfile>} a Promise that returns the original profile, for help in chaining calls
 	 */
-	public saveProfile(profile: IConnectionProfile, forceWritePlaintextPassword?: boolean): Promise<IConnectionProfile> {
+	public async saveProfile(profile: IConnectionProfile, forceWritePlaintextPassword?: boolean): Promise<IConnectionProfile> {
 		const self = this;
-		return new Promise<IConnectionProfile>((resolve, reject) => {
+		return new Promise<IConnectionProfile>(async (resolve, reject) => {
```

Comment:
Good opportunity here to apply the async/await style again - makes it much cleaner and easier to read since you can get rid of the whole wrapped promise stuff. (or good task to do separately as a code cleanup thing if you wanted)